### ISSUE TITLE
$-sign in code-block directive removes syntax highlighting

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#715]
(Original issue 1008 created by None on 2014-06-21T14:24:05.747935+00:00)

Sphinx ignores syntax highlighting when the code-block directive includes a dollar ($) sign:

```
#!rest

.. code-block:: sql

   SELECT * FROM v$sql;

```

gives no highlighting. When removing the dollar sign, the highlighting is back, which is not desired as the dollar sign is part of the code. Escaping the dollar sign does not work, as the code-block directive is verbatim.

First reported at Sphinx: https://bitbucket.org/birkenfeld/sphinx/issue/1494/sign-in-code-block-directive-removes:

```
#!python

$ echo SELECT * FROM v$sql; | pygmentize -f html -g
<div class="highlight"><pre><span class="n">SELECT</span> <span class="o">*</span> <span class="n">FROM</span> <span class="n">v</span><span class="err">$</span><span class="n">sql</span><span class="p">;</span>
</pre></div>

$ echo SELECT * FROM vSsql; | pygmentize -f html -g
<div class="highlight"><pre><span class="n">SELECT</span> <span class="o">*</span> <span class="n">FROM</span> <span class="n">vSsql</span><span class="p">;</span>
</pre></div>
```

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-07T14:02:04.194843+00:00)

Sphinx does not highlight the block if Pygments signals an error token.

Is the dollar sign a valid character in standard SQL names?

**Comment by Anteru** [comment]
(Original comment by None on 2014-10-07T17:20:56.144348+00:00)

Yes, it is in Oracle, as long as it isn't the first character of an object name.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-07T21:34:51.749439+00:00)

Closes #1008: allow $ in names for Oracle SQL.

→ <<cset ff014696f9b2>>
